spec/index.bs

```diff
@@ -1113,6 +1101,38 @@ Given an {{HTMLScriptElement}} (|script|), this algorithm performs the following
 
     If the algorithm threw an error, rethrow the error.
 
+## Get Trusted Types-compliant attribute value ## {#validate-attribute-mutation}
+To <dfn abstract-op export>get Trusted Types-compliant attribute value</dfn> on {{Attr}} |attribute| with {{Element}} |element| and {{TrustedType}} or a string |newValue|, perform the following steps:
+
+1. Let |expectedType| be null.
+1. If |attribute|'s <a for="Attr">namespace</a> is null or is a <a>HTML namespace</a>, find the row in the following table, where |element| is in the first column and |attribute|'s <a for="Attr">local name</a> is an <a>ASCII case-insensitive</a> match for any the string in the
+      second column. If a matching row is found, set |expectedType| to the value of the third column in that row.
+
+    <table>
+      <thead>
+        <tr><th>Element<th>Attribute local name</th><th>TrustedType<th>
+      <tbody>
+        <tr><td>{{HTMLIFrameElement}}<td>"srcdoc"<td>{{TrustedHTML}}
+        <tr><td>{{HTMLEmbedElement}}<td>"src"<td>{{TrustedScriptURL}}
+        <tr><td>{{HTMLScriptElement}}<td>"src"<td>{{TrustedScriptURL}}
+        <tr><td>{{HTMLObjectElement}}<td>"data"<td>{{TrustedScriptURL}}
+        <tr><td>{{HTMLObjectElement}}<td>"codebase"<td>{{TrustedScriptURL}}
```

Comment:
I guess this list doesn't include some elements/attributes because of the cross-document thingy. And still includes embed/object due to that being the status quo.

Reply:
The list mirrors the functionality on the IDL layer that's currently in the spec. It doesn't include the XSL-related things (e.g. https://github.com/w3c/trusted-types/issues/359), I'm not sure if those sinks could be guarded only by hooking on DOM attributes, it might require a different approach like disabling XSLT altogether.

Event handler attributes have an integration via HTML, and attributes supporting javascript: URLs are also handled in CSP integration. 

There might be xlink:href missing maybe for svg:script ? It's hard to say, pending https://github.com/w3c/svgwg/issues/196 in SVG, https://svgwg.org/svg2-draft/interact.html#ScriptElement is quite vague, but I guess we could add it here too?

Anything else I missed?

Reply:
Things I wasn't sure about:

1. frame src / iframe src
2. a href / area href / form action / * form action

I think 1 is out-of-scope and 2 is handled by CSP it sounds like.

The current list doesn't handle the SVG script element at all, so that needs to be added, yes. (It's not an HTMLScriptElement. In Chromium/WebKit they share a common class, but that's not a thing in specifications.)

Reply:
Yes, all of those sinks are handled by CSP blocking navigation to `javascript:`, and we don't control iframe/src beyond that.

> SVG
ACK.